src/pages/Users/UserProfile/components/UserTasksPreview.tsx

```diff
@@ -0,0 +1,108 @@
+import { Loader } from 'components/Loader';
+import { makeRequest } from 'hooks/makeRequest';
+import React, { FC, useEffect, useState } from 'react';
+import { ListItem } from 'types/ListTypes';
+import { UserType } from 'types/UserTypes';
+
+
+type UserTasksPreviewProps = {
+    userId: string;
+};
+
+const UserTasksPreview: FC<UserTasksPreviewProps> = ({ userId }) => {
+    const [user, setUser] = useState<UserType | null>(null);
+    const [tasks, setTasks] = useState<ListItem[]>([]);
+    const [loading, setLoading] = useState(true);
+
+    // useEffect(() => {
+    //     const fetchUserAndTasks = () => {
+    //         setLoading(true);
+
+    //         // Pobieramy dane użytkownika
+    //         makeRequest('GET', `/users/${userId}`)
+    //             .then(user => {
+    //                 setUser(user?.data);
+
+    //                 // Po pobraniu użytkownika, pobieramy zadania
+    //                 return makeRequest('GET', `/user/${userId}/tasks`);
+    //             })
+    //             .then(tasks => {
+    //                 setTasks(tasks?.data);
+    //                 setLoading(false);
+    //             })
+    //             .catch(err => {
+    //                 console.error('Error fetching user or tasks:', err);
+    //                 setLoading(false);
+    //             });
+    //     };
+
+    //     fetchUserAndTasks();
+    // }, [userId]);
+
+    useEffect(() => {
+        const fetchUser = () => {
+            setLoading(true)
+            makeRequest('GET', `/users/${userId}`)
+                .then(user => {
+                    setUser(user?.data)
+                })
+                .catch(err => {
+                    throw new Error('Error fetch user', err)
+                })
+        }
+
+        const fetchTasks = () => {
+            makeRequest('GET', `/user/${userId}/tasks`)
+                .then(tasks => {
+                    setTasks(tasks?.data)
+                    setLoading(false);
+                })
+                .catch(err => {
+                    throw new Error('Error fetching tasks', err)
+                })
+
+        }
+        fetchUser()
+        fetchTasks()
```

Comment:
po co pakujesz to w oddzielne funkcje skoro nie sa one asynchroniczne tak czy siak? Dodatkowo mozna by to wpakowac w Promise.all

Reply:
Nie sa asynchroniczne, bo nie uzywasz ani async ani await*